modules/mono/glue/GodotSharp/GodotSharp/Core/Callable.generics.cs

```diff
@@ -54,7 +54,7 @@ static void Trampoline(object delegateObj, NativeVariantPtrArgs args, out godot_
             ThrowIfArgCountMismatch(args, 1);
 
             ((Action<T0>)delegateObj)(
-                VariantConversionCallbacks.GetToManagedCallback<T0>()(args[0])
+                VariantUtils.ConvertTo<T0>(args[0])
```

Comment:
Interesting, does the analyzer not complain that `T0` is not annotated with `[MustBeVariant]`?

Reply:
Oh, I see, the analyzer is in `Godot.SourceGenerators` but in `GodotSharp` we are only using `Godot.SourceGenerators.Internal`. So the analyzer is not run for `GodotSharp` but it will be run in user projects. 

That's a shame because we missed the `[MustBeVariant]` attribute here and we need it if we want to prevent this:

```csharp
Callable.From((NonMarshableType a) => ...);
```

Reply:
We could extract the analyzer to its own project/assembly to re-use it.